### ISSUE TITLE
VectorGridProtobuf not working properly with LayerControl

### ISSUE THREAD

**Issue by giswqs** [python-visualization/folium#1716]
**Describe the bug**
The `VectorGridProtobuf` example works fine with the layer control. However, once the layer control is added, the vector tile layer can no longer be overlaid on the basemap layer. I am not sure if this a folium plugin problem or the upstream `leaflet.VectorGrid.Protobuf` issue. 

@iwpnd

```python
import folium
import folium.plugins as plugins

url = "https://tile.nextzen.org/tilezen/vector/v1/512/all/{z}/{x}/{y}.mvt?api_key=gCZXZglvRQa6sB2z7JzL1w"
m = folium.Map(location=[46.8, 8.2], zoom_start=14)
vc = plugins.VectorGridProtobuf(url, "vector tile")
vc.add_to(m)
folium.LayerControl().add_to(m)
m
```
![Peek 2023-01-31 11-45](https://user-images.githubusercontent.com/5016453/215827307-ed7776e5-c898-4111-a923-553654b8796d.gif)

**Expected behavior**

**Environment (please complete the following information):**
 - Python version: 3.9
 - folium version: 0.14.0

**Additional context**

**Possible solutions**
`folium.TileLayer` has the `overlay` argument that allows overlaying layer. Can `VectorGridProtobuf` have the same?

**Comment by iwpnd** [comment]
This looks to be on layer control as you're selecting either tiles or background and not both.

**Comment by giswqs** [comment]
The layer control shows the layers as radio buttons. You can only select on layer at a time, not both.

**Comment by giswqs** [comment]
Ideally, the vector tile layer should be listed with a checkbox next to it, not radio button. 

![Peek 2023-01-31 12-00](https://user-images.githubusercontent.com/5016453/215831025-2d67a218-33ff-49b8-9b1f-7aeba12b5ad6.gif)

**Comment by iwpnd** [comment]
Thank you for the pointer @giswqs, I found the issue and will be providing a PR with a fix asap

**Comment by iwpnd** [comment]
The issue is resolved with #1723 once merged.
